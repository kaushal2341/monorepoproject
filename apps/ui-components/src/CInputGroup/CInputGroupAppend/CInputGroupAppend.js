import React, {memo} from 'react';
import {InputGroup} from 'react-bootstrap';

const CInputGroupAppend = ({id, classes, append}) => {
    return (
        <InputGroup.Append
            id={id}
            className={classes}>
            {append && append.map((ap, index) => ap)}
        </InputGroup.Append>

    );
};
export default memo(CInputGroupAppend);