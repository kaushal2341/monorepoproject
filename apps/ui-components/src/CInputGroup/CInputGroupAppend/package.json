{
    "name": "@cogent/c-inputgroup-append",
    "main": "CInputGroupAppend.js",
    "version": "0.0.0",
    "license": "MIT",
    "private": true
}