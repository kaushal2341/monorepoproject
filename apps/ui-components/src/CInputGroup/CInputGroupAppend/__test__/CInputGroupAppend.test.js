import CInputGroupAppend from '../CInputGroupAppend';
import {InputGroup, Button} from 'react-bootstrap';

expect.addSnapshotSerializer(enzymeSerializer);
describe('CInputGroupAppend', () => {
    let wrapper;
    beforeEach(() => {
        wrapper = mount(
            <CInputGroupAppend
                id="appendInputGroup"
                classes="hello"
                append={''}
            />);
    });
    afterEach(() => {
        wrapper.unmount();
    });
    it('should be defined', () => {
        expect(wrapper).toBeDefined();
    });

    it('should have given props available', () => {
        const propsForAppend = ['id', 'classes', 'append'];
        const appendPropsAv = Object.keys(wrapper.props())
        propsForAppend.map((propsForApp, i) => {
            expect(propsForAppend).toContain(appendPropsAv[i])
        });
    });

    it('should have at least one element of input inside append element', () => {
        const a = [<InputGroup.Text key="addOn-1" id="basic-addOn3">
            https://example.com/users/
        </InputGroup.Text>,
            <Button key="button2" id="button2" variant="outline-secondary">Button</Button>
        ]
        wrapper.setProps({append: a})
        expect(wrapper.find('#basic-addOn3').length).toBeGreaterThan(1);
    });

    describe('Snapshot testing', () => {
        it('should match the previous snapshot', () => {
            expect(wrapper).toMatchSnapshot();
        })
    });

})