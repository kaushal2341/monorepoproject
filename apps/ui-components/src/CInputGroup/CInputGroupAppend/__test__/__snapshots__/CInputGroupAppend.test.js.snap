// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CInputGroupAppend Snapshot testing should match the previous snapshot 1`] = `
<Memo(CInputGroupAppend)
  append=""
  classes="hello"
  id="appendInputGroup"
>
  <InputGroupAppend
    className="hello"
    id="appendInputGroup"
  >
    <div
      className="hello input-group-append"
      id="appendInputGroup"
    />
  </InputGroupAppend>
</Memo(CInputGroupAppend)>
`;
